src/of-type.ts

```diff
@@ -22,27 +22,15 @@ import { castArray } from './utils'
  * )
  */
 export function ofType<
-  TSource extends AnyAction,
-  TActionCreator extends ActionCreator<TSource>,
->(
-  actionCreators: TActionCreator | ReadonlyArray<TActionCreator>
-): OperatorFunction<TSource, ReturnType<TActionCreator>>
-export function ofType<TSource extends AnyAction, TAction extends TSource>(
-  actions: TAction | ReadonlyArray<TAction>
-): OperatorFunction<TSource, TAction>
-export function ofType<
-  TSource extends AnyAction,
-  TType extends TSource['type']
->(
-  types: TType | ReadonlyArray<TType>
-): OperatorFunction<
-  TSource,
-  TSource extends Action<infer T> ? (T extends TType ? TSource : never) : never
->
-export function ofType(keys: ActionCreator<AnyAction> | AnyAction | string) {
-  const types = castArray(keys).map(key =>
-    typeof key === 'string' ? key : getType(key)
-  )
+  TSource extends string | AnyAction | ActionCreator<AnyAction>,
```

Comment:
In RxJS operators by convention, TSource is the input value type of stream and TSink is the output value type of stream. So let's stick to conventions. Following generics is needed:

- ```TSource extends AnyAction```
- ```TKey extends string | AnyAction | ActionCreator<AnyAction>```
- ```TSink extends TSource = ExtractAction<TKey, TSource>```

Reply:
I'm not a RxJS expert.
Thank you for teaching me!
i will fix as soon.

Reply:
completed! https://github.com/thebrodmann/deox/pull/133/commits/c517a553051e7a9ccaa0632adf50695b91facce8